src/actions/receive-profile.js

```diff
@@ -1145,7 +1145,20 @@ async function _extractJsonFromResponse(
   fileType: 'application/json' | null
 ): Promise<any> {
   try {
-    // Don't check the content-type, but attempt to parse the response as JSON.
+    // Don't check the content-type, check first if the response is gzipped,
+    // then attempt to parse the response as JSON.
+    // According to MDN, response.clone is not available on Android.
+    if (response.clone) {
+      const clonedResponse = response.clone();
+      const arrayBuffer = await clonedResponse.arrayBuffer();
+      const profileBytes = new Uint8Array(arrayBuffer);
+      // Check for the gzip magic number in the header.
+      if (profileBytes[0] === 0x1f && profileBytes[1] === 0x8b) {
+        const decompressedProfile = await decompress(profileBytes);
+        const textDecoder = new TextDecoder();
+        return JSON.parse(textDecoder.decode(decompressedProfile));
+      }
+    }
     const json = await response.json();
```

Comment:
As we discussed, it's probably a good idea to avoid the cloning, and do:
```js
        const textDecoder = new TextDecoder();
        return JSON.parse(textDecoder.decode(decompressedProfile));
```
in both cases. Just when we detect the gzip magic number, we decompress first.

I think we should try to mutualize more code with all other parts too, but that's out of scope for your patch.

### Testing
This is where all this code is tested:
https://github.com/firefox-devtools/profiler/blob/46989fadcf9099cdfa58e94d6f4f41e1bbfc862d/src/test/store/receive-profile.test.js#L832
As you see we're doing a crude mock of the `Response` object as it's not available in the node environment (and isn't implemented in jsdom that we use). One day we'll use a better library like `jest-fetch-mock` which includes an implementation for `Response`... (but again, out of scope for this patch).
In the mean time, you can add an implementation for `arrayBuffer` (and probably replace `json` in this case). You can also add another fake response that would return a compressed version of a profile:
```js
    const fetch200GzippedResponse = (({
      ok: true,
      status: 200,
      headers: {
        get: () => 'application/json',
      },
      arrayBuffer: () => compress(serializeProfile(_getSimpleProfile())),
    }: any): Response);
```
And then you can copy the first testcase in this `describe` block and use this response instead of `fetch200Response`.

I expect as well that you'll have to change some `json` into `arrayBuffer` in other locations too, because I'm quite sure this function is used when retrieving from store, and when retrieving profiles to compare (and possibly other cases too).